tests/h/search/query_test.py

```diff
@@ -165,6 +165,81 @@ def search(self, search, pyramid_request):
 
 
 class TestAnyMatcher(object):
+    def test_matches_uriparts(self, search, Annotation):
+        Annotation(target_uri="bar.com")
+        matched_ids = [Annotation(target_uri="foo.com").id,
+                       Annotation(target_uri="foo/bar.com").id]
+
+        result = search.run({"any": "foo"})
+
+        assert sorted(result.annotation_ids) == sorted(matched_ids)
+
+    def test_matches_quote(self, search, Annotation):
+        Annotation(target_selectors=[{'exact': 'selected bar text'}])
+        matched_ids = [Annotation(target_selectors=[{'exact': 'selected foo text'}]).id,
+                       Annotation(target_selectors=[{'exact': 'selected foo bar text'}]).id]
+
+        result = search.run({"any": "foo"})
+
+        assert sorted(result.annotation_ids) == sorted(matched_ids)
+
+    def test_matches_text(self, search, Annotation):
+        Annotation(text="bar is best")
+        matched_ids = [Annotation(text="foo is fun").id,
+                       Annotation(text="foo is bar's friend").id]
+
+        result = search.run({"any": "foo"})
+
+        assert sorted(result.annotation_ids) == sorted(matched_ids)
+
+    def test_matches_tags(self, search, Annotation):
+        Annotation(tags=["bar"])
+        matched_ids = [Annotation(tags=["foo"]).id,
+                       Annotation(tags=["foo", "bar"]).id]
+
+        result = search.run({"any": "foo"})
+
+        assert sorted(result.annotation_ids) == sorted(matched_ids)
+
+    def test_matches_all_fields(self, search, Annotation):
+        """
+        Any is expected to match any of the following fields:
+        quote, text, uri.parts, and tags.
+        """
+        Annotation(target_selectors=[{'exact': 'selected bar text'}])
+        Annotation(tags=["bar"])
+        Annotation(target_uri="bar.com")
+        Annotation(text="bar is best")
+        matched_ids = [Annotation(target_uri="foo.com").id,
+                       Annotation(target_selectors=[{'exact': 'selected foo bar text'}]).id,
+                       Annotation(text="foo is bar's friend").id,
+                       Annotation(tags=["foo", "bar"]).id]
+
+        result = search.run({"any": "foo"})
+
+        assert sorted(result.annotation_ids) == sorted(matched_ids)
+
+    def test_ors_any_matches(self, search, Annotation):
+        """
+        Any is expected to match any of the following fields;
+        quote, text, uri.parts, and tags
+        that contain any of the passed keywords.
+        """
+        Annotation(target_selectors=[{'exact': 'selected baz text'}])
+        Annotation(tags=["baz"])
+        Annotation(target_uri="baz.com")
+        Annotation(text="baz is best")
+        matched_ids = [Annotation(target_uri="foo/bar/baz.com").id,
```

Comment:
These matches all contain the term "foo" in one of the matched fields. In order to test the behaviour we expect to see here we'd need to have at least some annotations containing either "foo" OR "bar", but not both.